### PR TITLE
[Model][Backport] Remove mapWallet not needed call + stop treating coinbase differently

### PR DESCRIPTION
**Pull request by furszy**
Remove mapWallet not needed call + stop treating coinbases differently.

Coming from bitcoin@608bbccfb97e5ecb9083ac203642524a6e556e48.

### CODE REVIEW

File: src/qt/transactionrecord.cpp [hunk 1]
```diff
@@ -19,14 +19,10 @@
 
 /* Return positive answer if transaction should be shown in list.
  */
-bool TransactionRecord::showTransaction(const CWalletTx& wtx)
+bool TransactionRecord::showTransaction()
```
Comment (Warrows) [inline src/qt/transactionrecord.cpp]: I'd rather let the argument be or completely remove the function.

Comment (random-zebra) [reply src/qt/transactionrecord.cpp]: Agree on completely removing `TransactionRecord::showTransaction()` which should also make the private member `showTransaction` in `TransactionNotification` unnecessary.

Comment (furszy) [reply src/qt/transactionrecord.cpp]: done

**Comment by random-zebra** [comment]
needs rebase

**Pull request merged**